src/DependencyInjection/CustomMessageHeaderCompilerPass.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcingAdminBundle\DependencyInjection;
+
+use Patchlevel\EventSourcing\Metadata\Message\MessageHeaderRegistry;
+use Patchlevel\EventSourcingAdminBundle\Controller\DefaultController;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/** @interal */
+final class CustomMessageHeaderCompilerPass implements CompilerPassInterface
```

Comment:
I'm not entirely happy that we have to do it this way, but there's no other option at the moment. So it's fine for now. But I'd like to see at some point whether we can add something to the event sourcing lib to make it work better.

Reply:
Yeah, imho its related to this issue here: https://github.com/patchlevel/event-sourcing/issues/524